db/pinned_iterators_manager.h

```diff
@@ -16,7 +16,7 @@ namespace rocksdb {
 // PinnedIteratorsManager will be notified whenever we need to pin an Iterator
 // and it will be responsible for deleting pinned Iterators when they are
 // not needed anymore.
-class PinnedIteratorsManager {
+class PinnedIteratorsManager : public Cleanable {
```

Comment:
We can go either way but it would be great if we discuss it and have pros and cons listed. The pros of inheritance are:
1. Simplicity: no need to add the functions and the glue them to the Cleanable data member.
What would be the cons of this approach, or what would the pros of data member approach?